src/district0x/ui/events.cljs

```diff
@@ -79,71 +80,52 @@
    :on-success on-success
    :on-failure on-failure})
 
-
-(defn initialize-web3-instance
-  "Either re-uses a provided web3 provider, or creates a Web3 instance
-  for the re-frame `db`.
-
-  Returns the web3 instance to be included in the db side-effect"
-  [db]
-  (try-catch
-    (if (d0x-ui-utils/provides-web3?)
-      (new (aget js/window "Web3") (web3/current-provider (aget js/window "web3")))
-      (web3/create-web3 (:node-url db)))))
-
+(defn create-wallet-connect-web3
+  "Initiate web3 modal popup and allow user to choose a way how to
+  connect to Namebazaar.
+
+  The result of the web3 modal will be web3 provider which is used
+  to create new web3 instance (which is later saved into DB). This
+  handling is asynchronous and the function returns a promise."
+  []
+  ;; Clear the wallet connect local storage, which persists previous connection
+  ;; I've found the persisted connection to be extremely unreliable (transactions didn't work)
```

Comment:
I find the connection extremely unreliable from the first - I need page refreshes and reconnects to revive the ability to send transactions, and some transactions like Create auction offering and Delete offering I wasn't able to execute even if it's the first thing I do after reconnect.

Reply:
A second problem I was having trouble to get rid of was Nonce too low problem in wallet.

I replicated both problems in app.ens.domains though, which should be a flagship integration. Therefore I think there's general low graciousness in handling disconnects and some slowness in connection in walletconnect, which can't be solved by us.